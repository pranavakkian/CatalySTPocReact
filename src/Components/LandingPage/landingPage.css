.addbutton
{
    margin-top: 30px;
}
.viewbutton
{
    margin-top: 30px;
}