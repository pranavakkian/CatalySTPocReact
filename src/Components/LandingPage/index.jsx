import { useDispatch, useSelector } from 'react-redux';
import { memo } from 'react';
import { useHistory } from 'react-router-dom';
import { Button, Container, Header } from 'semantic-ui-react';
import { addColleaguePath } from '../../Constants';
import { handleIsLoggedIn, handleSetUserName } from '../../Redux/UserInfoReducer/Action';
import './landingPage.css';

const LandingPage = memo(() => {
  const history = useHistory();
  const userName = useSelector((state) => state.UserInfoReducer.userName);
  const dispatch = useDispatch();
  const signout = () => {
    dispatch(handleIsLoggedIn(false));
    dispatch(handleSetUserName(null));
  };
  return (
    <div>
      <Button floated="right" type="submit" onClick={signout}>Sign-out</Button>
      <Header as="h1" textAlign="center">
        Welcome
        {' '}
        {userName}
      </Header>
      <Container textAlign="center">
        <div className="addbutton">
          <Button onClick={() => history.push(addColleaguePath)}>Add Colleague</Button>
        </div>
        <div className="viewbutton">
          <Button>View Colleague</Button>
        </div>
      </Container>

    </div>
  );
});

export default LandingPage;
