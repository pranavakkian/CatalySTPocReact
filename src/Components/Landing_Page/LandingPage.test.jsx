import { render } from '@testing-library/react';
import LandingPage from './index';

describe('LandingPage', () => {
  test('Component rendered successfully', () => {
    const { container } = render(<LandingPage />);
    expect(container).toMatchSnapshot();
  });
});
