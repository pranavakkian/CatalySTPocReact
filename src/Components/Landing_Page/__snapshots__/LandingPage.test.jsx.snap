// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Component rendered successfully 1`] = `
<div>
  <div>
    <h1
      class="ui center aligned header"
    >
      Welcome to LandingPage
    </h1>
  </div>
</div>
`;
